src/parachain/tokens.ts

```diff
@@ -66,17 +66,17 @@ export class DefaultTokensAPI extends DefaultTransactionAPI implements TokensAPI
         currency: Currency<U>,
         id: AccountId
     ): Promise<MonetaryAmount<Currency<U>, U>> {
-        const currencyId = tickerToCurrencyIdLiteral(currency.ticker);
-        const account = await this.api.query.tokens.accounts(id, currencyId);
+        const currencyIdLiteral = tickerToCurrencyIdLiteral(currency.ticker);
+        const account = await this.api.query.tokens.accounts(id, newCurrencyId(this.api, currencyIdLiteral));
         return newMonetaryAmount(account.free.toString(), currency);
     }
 
     async balanceLocked<U extends CurrencyUnit>(
         currency: Currency<U>,
         id: AccountId
     ): Promise<MonetaryAmount<Currency<U>, U>> {
-        const currencyId = tickerToCurrencyIdLiteral(currency.ticker);
-        const account = await this.api.query.tokens.accounts(id, currencyId);
+        const currencyIdLiteral = tickerToCurrencyIdLiteral(currency.ticker);
+        const account = await this.api.query.tokens.accounts(id, newCurrencyId(this.api, currencyIdLiteral));
         return newMonetaryAmount(account.reserved.toString(), currency);
     }
```

Comment:
Minor, but could this duplicated code be consolidated? Other than the returned value these two methods are the same.

Reply:
One method returns the `reserved` while the other returns `free` balance. Good point though, I've refactored.